### PR TITLE
Dev feat

### PR DESCRIPTION
**Pull request by sergei-kabak-idf**
Realised bloc on call method

### CODE REVIEW

**Review by sergei-kabak-idf — COMMENTED** [review]
done

File: packages/presentation/lib/base/base_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  final _blocData = PalindromeData.init();
+
+  final _controllerStream = StreamController<PalindromeData>();
+
+  @protected
+  void updateState(bool? isPalindrome, String? inputPalindrome) {
+    if (!_controllerStream.isClosed) {
+      _blocData.call(
+          isPalindrome: isPalindrome, inputPalindrome: inputPalindrome);
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/base/base_bloc.dart]: formatting

File: packages/presentation/lib/screen/home_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+abstract class HomeBloc extends Bloc {
+  factory HomeBloc(
+    PalindromeUseCaseImpl palindromeUseCaseImpl,
+  ) =>
+      _HomeBloc(palindromeUseCaseImpl);
+
+  void checkPalindrome();
+
+  void setString(String text);
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/home_bloc.dart]: for the future - naming setString says about nothing. think over renaming it

File: packages/presentation/lib/screen/home_data.dart [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+  factory PalindromeData.init() => PalindromeData(false, '');
+
+  void call({
+    required bool? isPalindrome,
+    required String? inputPalindrome,
+  }) {
+    if (isPalindrome != null) this.isPalindrome = isPalindrome;
+    if (inputPalindrome != null) this.inputPalindrome = inputPalindrome;
+  }
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/home_data.dart:15]: what is this method needed for?

File: packages/presentation/lib/screen/home_page.dart [hunk 4]
```diff
@@ -12,31 +11,52 @@ class HomePage extends StatefulWidget {
...
         ),
         floatingActionButtonLocation: FloatingActionButtonLocation.centerFloat,
         floatingActionButton: _showButtonWidget(),
-      ),
-    );
-  }
+      );
+
+  Widget _showButtonWidget() => FloatingActionButton(
+        onPressed: () => checkPalindrome(),
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/home_page.dart]: if i'm not mistaken, u can use it in the following way:

`onPressed: checkPalindrome`

just fyi

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]
Look over `StreamPlatformStackContent` implementation which I showed u. There's no specifying body directly in widget. in that variant callback which is passed from screen is called. Refactor please in this way

File: packages/presentation/lib/base/base_bloc.dart (hunk 1 above)
Comment (sergei-kabak-idf) [reply packages/presentation/lib/base/base_bloc.dart]: done

File: packages/presentation/lib/screen/home_bloc.dart (hunk 2 above)
Comment (sergei-kabak-idf) [reply packages/presentation/lib/screen/home_bloc.dart]: renamed

File: packages/presentation/lib/screen/home_data.dart (hunk 3 above)
Comment (sergei-kabak-idf) [reply packages/presentation/lib/screen/home_data.dart:15]: updating properties in the instance PalindromeData classes.

File: packages/presentation/lib/screen/home_page.dart (hunk 4 above)
Comment (sergei-kabak-idf) [reply packages/presentation/lib/screen/home_page.dart]: done

**Review by sergei-kabak-idf — COMMENTED** [review]
fixed

**Pull request closed without merging**